tests/chat/test_chat.py

```diff
@@ -1,54 +1,75 @@
+import os
 import tempfile
 import time
+import mongomock
 from channels.testing import ChannelsLiveServerTestCase
+from mongoengine import connect, disconnect
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
 from selenium.webdriver.chrome.service import Service
 from selenium.webdriver.common.action_chains import ActionChains
 from selenium.webdriver.common.keys import Keys
 from webdriver_manager.chrome import ChromeDriverManager
+from users.documents import UserDocument, UserRoleDocument, UserRoleEnum
 
+os.environ['DJANGO_SETTINGS_MODULE'] = 'core.test_settings'
```

Comment:
Consider moving environment setup into a test-specific entrypoint or pytest fixture for clarity and isolation.